CDCgov/prime-reportstream
# Better AOE Processing

This PR addresses an issue where the AOE processing for the monkeypox messages was missing any OBX segments placed under the SPM segment which is legal and valid HL7, but does not match the assumptions we made using the terser.

This logic now converts the message we're working with to an `ORU_R01` message and then walks both areas where OBX's could live and looks at OBX-3 to see if they match the AOE question code we're looking for. Then the code actually attempts to coerce the OBX-5 value into its correct datatype (e.g. `DT`, `NM`, etc) and extract the identifier value we want.

Test Steps:
1. Pass a message through that has AOE OBX segments under the SPM value and see if they are correctly parsed.
2. Run the unit tests and integration tests

## Changes
- Moved the AOE processing into the companion object because it's static
- Changed the way it processes to not use the terser
- Created a second function to parse the OBX-5 object based on the data type OR on its raw string value

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue
